q2_types/feature_data/_format.py

```diff
@@ -394,6 +394,52 @@ def validate(self, *args):
     'BLAST6DirectoryFormat', 'blast6.tsv', BLAST6Format)
 
 
+class UNIXListFormat(model.TextFileFormat):
+    def _validate_(self, level):
+        # any file with lines will be valid
+        return True
+
+    def to_list(self):
+        with self.open() as fh:
+            return [s.strip() for s in fh]
+
+
+class IDMetadataFormat(model.TextFileFormat):
```

Comment:
We didn't have the `ImmutableMetadata` type/format when we created this - we could probably update genome-sampler to not use this anymore, but I think we'll need to keep the type/format around for working with previously generated artifacts.